cassiopeia/solver/GreedySolver.py

```diff
@@ -46,18 +51,26 @@ def __init__(
         character_matrix: pd.DataFrame,
         missing_char: str,
         meta_data: Optional[pd.DataFrame] = None,
-        priors: Optional[Dict] = None,
+        priors: Optional[Dict[int, Dict[int, float]]] = None,
+        weights: Optional[Dict[int, Dict[int, float]]] = None,
     ):
 
         super().__init__(character_matrix, missing_char, meta_data, priors)
+        self.weights = weights
+        if priors:
+            self.priors = priors
+            if not weights:
+                self.weights = solver_utilities.negative_log_prob_weights(
```

Comment:
Is this a good design decision? Do you imagine doing other fancy things than just applying `np.log10` to all priors?

Reply:
Yeah Robert and I have floated a couple of ideas for other ways to utilize priors, like taking a 1/(m * q{j}) weight for use in similarity. I think it would be helpful to have this going down the line

Reply:
Cool, let's change this function to `transform_priors` and here we can just pass in -1*np.log10.